python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+    )
+    outputCatalog = pipeBase.connectionTypes.Output(
+        doc=("Source table with stars used in fit, along with residuals in pixel coordinates and tangent "
+             "plane coordinates and chisq values"),
+        name="WCSFit_usedStars",
+        storageClass="DataFrame",
+        dimensions=("instrument", "skymap", "tract"),
+    )
+
+
+class WCSFitConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=WCSFitConnections):
+    """Configuration for WCSFitTask"""
+    sourceSelector = sourceSelectorRegistry.makeField(
+        doc="How to select sources for cross-matching",
+        default="science"
+    )
+    referenceSelector = pexConfig.ConfigurableField(
+        target=ReferenceSourceSelectorTask,
+        doc="How to down-select the loaded astrometry reference catalog.",
+    )
+    matchRadius = pexConfig.Field(
+        doc="Matching tolerance between associated objects (arcseconds)",
+        dtype=float,
+        default=1.0
+    )
+    minMatches = pexConfig.Field(
+        doc="Number of matches required to keep a source object",
+        dtype=int,
+        default=2
+    )
+    allowSelfMatches = pexConfig.Field(
+        doc="Allow multiple object matches from the same viist",
+        dtype=bool,
+        default=False
+    )
+    sourceFluxType = pexConfig.Field(
+        dtype=str,
+        doc="Source flux field to use in source selection and to get fluxes from the catalog.",
+        default='apFlux_12_0'
+    )
+    systematicError = pexConfig.Field(
+        dtype=float,
+        doc="Systematic error padding for the science images (marcsec)",
+        default=0.0
+    )
+    referenceSystematicError = pexConfig.Field(
+        dtype=float,
+        doc="Systematic error padding for the reference catalog (marcsec)",
+        default=0.0
+    )
+    modelComponents = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from pixels to sky, in order of their application",
+        default=["INSTRUMENT/DEVICE", "EXPOSURE"]
+    )
+    deviceModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from detector pixels to intermediate frame",
+        default=["BAND/DEVICE/poly"]
+    )
+    exposureModel = pexConfig.ListField(
+        dtype=str,
+        doc="List of mappings to apply to transform from intermediate frame to sky coordinates",
+        default=["EXPOSURE/poly"]
+    )
+    devicePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of device polynomial model",
+        default=4
+    )
+    exposurePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of exposure polynomial model",
+        default=6
+    )
+    useProperMotion = pexConfig.Field(
+        dtype=bool,
+        doc="Fit the proper motions of the objects",
+        default=False
+    )
+    verbose = pexConfig.Field(
+        dtype=int,
+        doc="Verbosity level=0, 1, or 2",
```

Comment:
If gbdes is using `print()` and not `logging` then there are going to be all sorts of wrinkles with capturing the output in batch (we capture logs but where stdout goes depends on the workflow system).

Is it possible to map these verbosity levels to WARN, INFO, VERBOSE log levels and so derive the verbosity level based on the logging level that is active for your task code? Ideally gbdes would use logging properly (although you might still have to work out how to set the default log level for it since we don't set non-lsst logger levels by default.

Reply:
All the output from `gbdes` is from c++ and I think it's all from `std::cerr`. It doesn't currently use any logger, it just prints to standard error depending on the verbosity level (which could be matched to WARN, INFO, etc).  @timj, do you know if there is any way to capture this properly without using `lsst.log`?

Reply:
Standard error is going to be caught by the pipeline infrastructure and will end up somewhere but it won't be stored in the butler logs or available to the real-time logging system. I think the C++ stderr is not the same as the python stderr so capturing it and forwarding it to a logger is a bit tricky (I think @ktlim tried to get something working for that when we used to use lsst.log in the past in jupyter notebooks). The best way is to start using the C++ lsst.log package in gbdes.

Reply:
I removed the config parameter and now set `verbose` for `gbdes` from `self.log.getEffectiveLevel()`. I'm going to make a jira ticket for improving the output in `gbdes`.